zebedee-cms/src/main/java/com/github/onsdigital/zebedee/session/store/SessionsThreadLocalImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.github.onsdigital.zebedee.session.store;
+
+import com.github.onsdigital.zebedee.session.store.exceptions.*;
+
+import com.github.onsdigital.exceptions.*;
+
+import com.github.onsdigital.impl.*;
+import com.github.onsdigital.interfaces.*;
+import javax.servlet.http.HttpServletRequest;
+import com.github.onsdigital.JWTHandlerImpl;
+
+/**
+ * This method creates a ThreadLocal for the validated jwt 
+ * from a HTTP request and appropriate key 
+ * as part of implementing dp-identity-api
+ */
+
+public class SessionsThreadLocalImpl implements SessionsThreadLocal {
+
+    private static HttpServletRequest request;
+    private static String secretKey;
+    private static ThreadLocal<UserDataPayload> store =
+            new ThreadLocal<>();
+
+    private JWTHandler jwtHandler = new JWTHandlerImpl();
+
+
+    public SessionsThreadLocalImpl(HttpServletRequest request, String secretKey) {
+        this.request = request;
+        this.secretKey = secretKey;
+    }
+
+    @Override
+    public void store(HttpServletRequest request, String secretKey ) throws SessionsStoreException {
+       /**
+        * Validates the Http Request header for null or empty 
+        * HttpServletRequest method for getHeader returns null if the request does not have a header of that name
+        */  
+
+        if ( request.getHeader("Authorization") == null || 
+                request.getHeader("Authorization").isEmpty() ) {
+                    throw new SessionsRequestException("Request does not have Authorization in Header");
+                };
+
+        /**
+         * validates the Key argument for null or empty string
+         *  */        
+        if (secretKey == null || 
+                secretKey.isEmpty()
+         ) {
+            throw new SessionsKeyException("Key is invalid");
+         }
```

Comment:
Formatting. Also as `secretKey` is a string it could be empty as well as null so can you update this to:
```
if (StringUtils.isEmpty(secretKey) {
    throw new SessionsKeyException("secret key value expected but was null or empty");
}
```
I've updated the exception message here to give a more precise description of the nature of the error. This will make it much easier for someone to understand and debug if they run into this exception.

Reply:
Amended message to "Secret key value expected but was null or empty." 
also added assertion to test for message 
"Request does not have Authorization in Header."
Added assertion for message